### PR TITLE
Improvements to container_builds role

### PR DESCRIPTION
**Pull request by raukadah**
Improvements to containers_build role

- Improve default role vars 
- Added separate task to tag and push containers using buildah
- Added missing exclude containers for master

Signed-off-by: Chandan Kumar <raukadah@gmail.com>

### CODE REVIEW

**Comment by rlandy** [comment]
works for me - containers pushed

**Review by rlandy — APPROVED** [review]
Works on test patches

File: ci/roles/build_containers/defaults/main.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
-# Path of the authentication file. Use REGISTRY_AUTH_FILE environment variable to override. (default: /root/containers/auth.json)
-# authfile_path
+# Path of the authentication file. Use REGISTRY_AUTH_FILE environment variable to override.
+authfile_path: ${XDG_RUNTIME_DIR}/containers/auth.json
```
Comment (marios) [inline ci/roles/build_containers/defaults/main.yaml:3]: what/where is setting this XDG_RUNTIME_DIR ?

**Review by marios — COMMENTED** [review]
changes seem reasonable

File: ci/roles/build_containers/defaults/main.yaml (hunk 1 above)
Comment (raukadah) [reply ci/roles/build_containers/defaults/main.yaml:3]: ${XDG_RUNTIME_DIR} is a env variable. The path automatically gets created on doing podman/buildah login.
More info is here: https://docs.podman.io/en/latest/markdown/podman-login.1.html#description
```
❯ echo ${XDG_RUNTIME_DIR}
/run/user/1000
```

Comment (marios) [reply ci/roles/build_containers/defaults/main.yaml:3]: thank you

**Pull request merged**